### PR TITLE
chore(docs): Fix link in release notes

### PR DESCRIPTION
**Pull request by bruceg**

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Thanks!

File: website/content/en/highlights/2024-05-07-0-38-0-upgrade-guide.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Vector's 0.38.0 release includes a **breaking change**:
 
 and **deprecations**:
 
-1. [Deprecation of undefined environment variable warnings](#strict-env-vars)
+1. [Deprecation of path coalescing operator](#path-coalescing)
```
Comment (maycmlee) [inline website/content/en/highlights/2024-05-07-0-38-0-upgrade-guide.md:18]:
```suggestion
1. [Deprecation of the path coalescing operator](#path-coalescing)
```

**Review by maycmlee — APPROVED** [review]
Just a small suggestion, but approving.

**Comment by jszwedko** [comment]
Merging manually to get this out since the integration tests timed out.

**Pull request merged**